src/main.zig

```diff
@@ -1380,6 +1386,12 @@ fn buildOutputType(
         }
     }
 
+    if (cross_target.isNativeOs() and std.Target.current.isDarwin()) {
+        if (syslibroot == null) {
+            syslibroot = try std.zig.system.getSDKPath(arena);
+        }
+    }
```

Comment:
I think we should move this code into Compilation.zig so that we can do it only in the path where `use_lld` is `true`. No sense in wasting precious milliseconds executing `xcrun` when your self-hosted linker code does not need this information.

https://github.com/ziglang/zig/blob/909aae8153e7ac71197bc9f864e4ceb2793317f2/src/Compilation.zig#L443

After we have this flag you could put this logic there. Also if you gate it behind `build_options.have_llvm` which is a comptime value, then we won't even have the code bloat of the ability to execute `xcrun` for non-llvm-enabled builds.

Reply:
Nice! This is an excellent suggestion!